packages/core/src/node/cluster.spec.ts

```diff
@@ -6,79 +6,60 @@
  */
 
 // tslint:disable:no-console
-import { assert, use } from 'chai';
-use(require('chai-as-promised'));
+import * as path from 'path';
 import * as cluster from 'cluster';
-import { clusterRemoteMasterProcessFactory, ServerProcess } from './cluster';
 import { MasterProcess } from './cluster/master-process';
 
 process.on('unhandledRejection', (reason, promise) => {
     throw reason;
 });
 
-if (cluster.isMaster) {
-    describe('master-process', () => {
+describe('master-process', () => {
 
-        function prepareWorkerTest(test: string) {
-            // https://github.com/Microsoft/vscode/issues/3201
-            const execArgv = process.execArgv.reduce((result, arg) => {
-                result.push(arg.replace('-brk', ''));
-                return result;
-            }, [] as string[]);
-            const testSettings: cluster.ClusterSettings = {
-                execArgv,
-                args: [...process.argv.slice(2), '--grep', test],
-                // https://github.com/Microsoft/vscode/issues/19750
-                stdio: ['ipc', 'pipe', 'pipe']
-            };
-            cluster.setupMaster(testSettings);
-        }
-
-        const forwardWorkerTestOutput = (worker: cluster.Worker) => {
-            worker.process.stdout.pipe(process.stdout);
-            worker.process.stderr.pipe(process.stderr);
+    function prepareTestWorker(job: string) {
+        const testSettings: cluster.ClusterSettings = {
+            exec: path.resolve(__dirname, '../../lib/node/test/cluster-test-worker.js'),
```

Comment:
I am just interested: does this work well with `ts-node`? I would expect, `ts-node` is smart enough to figure out the output directory, then with some mapping, it can calculate the module location, but I am just guessing. That's why I've asked.

Reply:
no, it does not, `cluster-test-worker.ts` should be compiled
We will need to fork ts-node here with the proper config. I've tried and failed, feel free to look into it. The important point that the output, especially errors, should be piped to the test output, not swallowed by ts-node.